cluster/src/main/java/org/apache/iotdb/cluster/service/TSServiceClusterImpl.java

```diff
@@ -53,17 +66,50 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TSServiceClusterImpl.class);
 
-  private ThreadLocal<NonQueryExecutor> nonQueryExecutor = new ThreadLocal<>();
-  private ThreadLocal<QueryMetadataExecutor> queryMetadataExecutor = new ThreadLocal<>();
+  private ClusterQueryProcessExecutor queryDataExecutor = new ClusterQueryProcessExecutor();
```

Comment:
It's unnecessary to make `NonQueryExecutor` and `QueryMetadataExecutor` be ThreadLocal. Only some fields in the class need to be ThreadLocal, it will waste resource to do like before.